packages/next/src/server/typescript/rules/client-boundary.ts

```diff
@@ -75,19 +86,14 @@ const clientBoundary = {
             ts.isConstructorTypeNode(typeDeclarationNode) ||
             ts.isClassDeclaration(typeDeclarationNode)
           ) {
-            // There's a special case for the error file that the `reset` prop is allowed
-            // to be a function:
-            // https://github.com/vercel/next.js/issues/46573
-            if (!(isErrorFile || isGlobalErrorFile) || propName !== 'reset') {
```

Comment:
Don't we need to keep this if people use class components in their `error.tsx`?

Reply:
The `reset` prop can only be a function type, so this check is not needed in this branch. For the other two scenarios I added examples to `test/development/typescript-plugin/app/client.tsx`, which I do find a bit academic, to be honest.